packages/start/config/server-runtime.jsx

```diff
@@ -33,11 +33,12 @@ async function deserializeStream(id, response) {
     }
   }
 
-  const result = await reader.read();
-  if (result.done) {
-    throw new Error("Unexpected end of body");
+  let serialized = ""
+  while (true) {
+    const line = await reader.read();
+    if (line.done) break;
```

Comment:
This is the problematic line. It's awaiting until the stream is done. Server functions are supposed to be non-blocking, and this entire mechanism is already implemented in the `pop` function declared just a few lines before this.